src/subsplit_dag.cpp

```diff
@@ -691,38 +701,52 @@ void SubsplitDAG::BuildDAGEdgesFromEdgeIndexer(BitsetSizeMap &edge_indexer) {
            "All edges should be bitsets with size 3 times taxon_count_.");
     const auto parent_id = GetDAGNodeId(edge.PCSPGetParentSubsplit());
     const auto child_id = GetDAGNodeId(edge.PCSPGetChildSubsplit());
-    SafeInsert(dag_edges_, {parent_id, child_id}, index);
+    auto left = GetDAGNode(parent_id).GetLeftLeafward();
+    auto clade = std::find(left.begin(), left.end(), child_id) != left.end()
+                     ? Clade::Left
+                     : Clade::Right;
+    storage_.AddLine({index, parent_id, child_id, clade});
   }
 }
 
 void SubsplitDAG::AddLeafSubsplitsToDAGEdgesAndParentToRange() {
   for (size_t node_id = 0; node_id < taxon_count_; node_id++) {
-    const auto current_bitset(GetDAGNode(node_id)->GetBitset());
+    const auto current_bitset(GetDAGNode(node_id).GetBitset());
     IterateOverRootwardEdges(
         GetDAGNode(node_id),
-        [this, current_bitset](const bool rotated, const SubsplitDAGNode *node) {
-          SafeInsert(parent_to_child_range_, node->GetBitset(rotated),
+        [this, current_bitset](const bool rotated, SubsplitDAGNode node) {
+          SafeInsert(parent_to_child_range_, node.GetBitset(rotated),
                      {EdgeCountWithLeafSubsplits(), EdgeCountWithLeafSubsplits() + 1});
-          SafeInsert(dag_edges_, {node->Id(), GetDAGNodeId(current_bitset)},
-                     EdgeCountWithLeafSubsplits());
+          storage_.AddLine({EdgeCountWithLeafSubsplits(), node.Id(),
+                            GetDAGNodeId(current_bitset),
+                            rotated ? Clade::Left : Clade::Right});
         });
   }
 }
 
+void SubsplitDAG::StoreEdgeIds() {
+  for (auto i : storage_.GetLines()) {
```

Comment:
I would suggest a more descriptive variable name here. And we don't want a const ref here?